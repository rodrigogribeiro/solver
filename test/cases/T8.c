typedef struct T { int x; char y; } A;
void f() {
    A a;
    T i = a.x;
}
