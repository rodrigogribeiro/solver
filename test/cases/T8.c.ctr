typedef A as struct T {int x; int y; },
 def f : (void)  in
  exists alpha1.
   typedef A as alpha1, def a : alpha1 in
    exists alpha2.
     exists alpha3.
      exists alpha4.
       typeof(a) = alpha3, has (alpha3, x : alpha4), alpha2 = alpha4,
        typedef T as alpha2, def i : alpha2 in
         alpha2 = T,
          alpha1 = A
