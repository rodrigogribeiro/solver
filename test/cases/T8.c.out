typedef struct T {int x;int y;} A;
